packages/@lwc/babel-plugin-component/src/__tests__/wire-decorator.spec.js

```diff
@@ -40,6 +40,73 @@ describe('Transform property', () => {
                       },
                       static: {
                         key2: ["fixed", "array"]
+                      },
+                      config: function(host) {
+                        return {
+                          key2: ["fixed", "array"],
+                          key1: host.prop1 != null ? host.prop1 : undefined
+                        };
+                      }
+                    }
+                  }
+                });
+
+                export default _registerComponent(Test, {
+                  tmpl: _tmpl
+                });
+`,
+            },
+        }
+    );
+
+    pluginTest(
+        'transforms parameters with multiple levels deep',
+        `
+        import { wire } from 'lwc';
+        import { getFoo } from 'data-service';
+        export default class Test {
+            @wire(getFoo, { key1: "$prop1.prop2.prop3.prop4", key2: ["fixed", 'array']})
+            wiredProp;
+        }
+    `,
+        {
+            output: {
+                code: `
+                import { registerDecorators as _registerDecorators } from "lwc";
+                import _tmpl from "./test.html";
+                import { registerComponent as _registerComponent } from "lwc";
+                import { getFoo } from "data-service";
+
+                class Test {
+                  constructor() {
+                    this.wiredProp = void 0;
+                  }
+                }
+
+                _registerDecorators(Test, {
+                  wire: {
+                    wiredProp: {
+                      adapter: getFoo,
+                      params: {
+                        key1: "prop1.prop2.prop3.prop4"
+                      },
+                      static: {
+                        key2: ["fixed", "array"]
+                      },
+                      config: function(host) {
+                        return {
+                          key2: ["fixed", "array"],
+                          key1:
+                            host.prop1 != null
```

Comment:
@caridy i believe we could replace this with a condition:
```js
host.prop1 != null && host.prop1.prop2 != null && host.prop1.prop2.prop3 != && host.prop1.prop2.prop3.prop4 != null
  ? host.prop1.prop2.prop3.prop4
  : undefined
```

Reply:
You can drop the end ternary operator. This expression is equivalent:

```js
host.prop1 != null && host.prop1.prop2 != null && host.prop1.prop2.prop3 != null && host.prop1.prop2.prop3.prop4 != null && host.prop1.prop2.prop3.prop4
```

Reply:
I don't think those are equivalent because they evaluate to `false` instead of `undefined` when a piece of the path is null or undefined. which is the same bug we have today in wire.

Reply:
Yes you are right.